File: Algorithm/Baekjoon/Python/브루트 포스/체스판 다시 칠하기.py
import math

n, m = map(int, input().split(" "))

sample = ["WBWBWBWB", "BWBWBWBW"]
board = []
for i in range(n):
    board.append(input())

answer = math.inf
for i in range(n - 7):
    tmp = board[i : i + 8]
    for j in range(m - 7):
        for c in range(2):
            cnt = 0
            for t in range(8):
                tmp2 = tmp[t][j : j + 8]
                if (t + c) % 2 == 0:
                    correct = sample[0]
                else:
                    correct = sample[1]

                for check in range(8):
                    if correct[check] != tmp2[check]:
                        cnt += 1
            answer = min(answer, cnt)

print(answer)

File: Algorithm/Baekjoon/Java/브루트포스/체스판_다시_칠하기.java
package Algorithm.Baekjoon.Java.브루트포스;

import java.util.*;

public class 체스판_다시_칠하기 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        String[] init = sc.nextLine().split(" ");
        int n = Integer.parseInt(init[0]);
        int m = Integer.parseInt(init[1]);
        String[] board = new String[n];
        String[] sample = { "WBWBWBWB", "BWBWBWBW" };
        for (int i = 0; i < n; i++) {
            board[i] = sc.nextLine();
        }
        sc.close();

        int answer = 64;
        for (int i = 0; i < n - 7; i++) {
            String[] tmp = Arrays.copyOfRange(board, i, i + 8);
            for (int j = 0; j < m - 7; j++) {
                for (int c = 0; c < 2; c++) {
                    int cnt = 0;
                    for (int t = 0; t < 8; t++) {
                        String tmp2 = tmp[t].substring(j, j + 8);
                        String correct = sample[(t + c) % 2];

                        for (int check = 0; check < 8; check++) {
                            if (tmp2.charAt(check) != correct.charAt(check)) {
                                cnt++;
                            }
                        }

                    }
                    answer = Math.min(answer, cnt);
                }
            }
        }

        System.out.println(answer);
    }
}
